src/utils/filterPeople.ts

```diff
@@ -0,0 +1,48 @@
+import { GenderFilters, Person, SearchQueries } from '../types';
+
+export function filterPeople(
+  people: Person[],
+  params: URLSearchParams,
+): Person[] {
+  let filtered = [...people];
```

Comment:
Instead of creating a copy of the `people` array, you can directly use the `filter` method on it. This would save memory as you won't be creating an unnecessary copy of the array.